widgets/topology/src/Topology.tsx

```diff
@@ -97,34 +99,18 @@ export default class Topology extends React.Component<TopologyProps, TopologySta
         const { blueprintId, configuration, data } = this.props;
         const { expandedDeployments, expandedTerraformNodes } = this.state;
 
-        if (
+        const topologyRestartAndUpdateRequired =
             configuration !== prevProps.configuration ||
             blueprintId !== prevProps.blueprintId ||
-            !_.isEqual(data.icons, prevProps.data.icons)
-        ) {
-            this.startTopology();
-        }
+            !_.isEqual(data.icons, prevProps.data.icons);
 
-        if (
-            data.blueprintDeploymentData !== prevProps.data.blueprintDeploymentData ||
+        const topologyUpdateRequired =
+            !_.isEqual(data.blueprintDeploymentData, prevProps.data.blueprintDeploymentData) ||
             expandedDeployments !== prevState.expandedDeployments ||
-            expandedTerraformNodes !== prevState.expandedTerraformNodes
-        ) {
-            const isFirstTimeLoading = this.topologyData === null;
-            const oldTopologyData = this.topologyData;
-            this.topologyData = this.buildTopologyData();
-            this.topology.setLoading(false);
-
-            if (_.isEmpty(data.blueprintDeploymentData)) {
-                this.topology.setTopology(this.topologyData, {});
-            } else if (isFirstTimeLoading || isNodesChanged(oldTopologyData.nodes, this.topologyData.nodes)) {
-                const { layout } = data;
-                this.topology.setTopology(this.topologyData, layout);
-                if (isFirstTimeLoading) this.topology.setScale(_.get(layout, 'scaleInfo'));
-            } else {
-                this.topology.refreshTopologyDeploymentStatus(this.topologyData);
-            }
-        }
+            expandedTerraformNodes !== prevState.expandedTerraformNodes;
+
+        if (topologyRestartAndUpdateRequired) this.startTopology();
+        if (topologyRestartAndUpdateRequired || topologyUpdateRequired) this.updateTopology();
```

Comment:
```suggestion
        const topologyRestartRequired =
            configuration !== prevProps.configuration ||
            blueprintId !== prevProps.blueprintId ||
            !_.isEqual(data.icons, prevProps.data.icons);

        const topologyUpdateRequired =
            topologyRestartRequired ||
            !_.isEqual(data.blueprintDeploymentData, prevProps.data.blueprintDeploymentData) ||
            expandedDeployments !== prevState.expandedDeployments ||
            expandedTerraformNodes !== prevState.expandedTerraformNodes;

        if (topologyRestartRequired) this.startTopology();
        if (topologyUpdateRequired) this.updateTopology();
```
 
With that approach we could correlate one variable with a corresponding functionality (restart is for restarting and updating is for updating 😁), making the function more readable and easier to understand 🎉 

(I know that by adding extra condition to one of the variables, it makes the 'if`ology' bigger, but as it already had a few conditions, I think that we shouldn't be worried about it)

WDYT? 🤔

Reply:
Of course, good idea. Applied.